ui/raidboss/data/07-dt/trial/valigarmanda-ex.ts

```diff
@@ -0,0 +1,760 @@
+      type: 'MapEffect',
+      // 00020001 - cleaves SW half (front/right safe)
+      // 00200010 - cleaves NE half (back/left safe)
+      netRegex: { flags: ['00020001', '00200010'], location: '03' },
+      run: (data, matches) => {
+        if (matches.flags === '00020001')
+          data.avalancheSafe = 'frontRight';
+        else
+          data.avalancheSafe = 'backLeft';
+      },
+    },
+    {
+      id: 'Valigarmanda Ex Big AOE + Avalanche',
+      type: 'StartsUsing',
+      // no cast bar, paired with an avalanche
+      // 8FC6: Susurrant Breath (conal)
+      // 8FCA: Slithering Strike (out)
+      // 8FCE: Strangling Coil (donut)
+      netRegex: { id: ['8FC6', '8FCA', '8FCE'], source: 'Valigarmanda' },
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        // these casts also happen in the final (no-avalanche) aoe mechanic
+        // so use data.avalancheSafe to prevent this trigger from mis-firing
+        if (data.avalancheSafe === undefined)
+          return;
+
+        // we can use backLeft/frontRight output as/is for donut and out,
+        // but for cone, we'll need to tweak it
+        let safe: 'backLeft' | 'frontRight' | 'coneNWSafe' | 'coneNESafe';
+        if (matches.id === '8FC6')
+          safe = data.avalancheSafe === 'backLeft' ? 'coneNWSafe' : 'coneNESafe';
+        else
+          safe = data.avalancheSafe;
+
+        const safeOutput = output[safe]!();
+
+        let typeOutput;
+        if (matches.id === '8FC6')
+          typeOutput = output.cone!();
+        else if (matches.id === '8FCA')
+          typeOutput = output.out!();
+        else
+          typeOutput = output.donut!();
+
+        return output.combo!({ type: typeOutput, safe: safeOutput });
+      },
+      run: (data) => delete data.avalancheSafe,
+      outputStrings: {
+        ...bigAoeOutputStrings,
+        backLeft: {
+          en: 'Be Back/Left',
+        },
+        frontRight: {
+          en: 'Be Front/Right',
+        },
+        coneNWSafe: {
+          en: 'NW Safe',
+        },
+        coneNESafe: {
+          en: 'NE Safe',
+        },
+        unknown: {
+          en: 'Dodge Avalanche',
+        },
+        combo: {
+          en: '${type} - ${safe}',
+        },
+      },
+    },
+    {
+      // Safe corner is opposite the northmost sphere
+      // NOTE: Have not seen any logs with stale position data, but if its an issue,
+      // this can be changed to a `getCombatants` call.
+      id: 'Valigarmanda Ex Ice Arcane Sphere Safe',
+      type: 'AddedCombatant',
+      netRegex: { name: 'Arcane Sphere' },
+      condition: (data) => data.phase === 'ice',
+      alertText: (data, matches, output) => {
+        const posY = parseFloat(matches.y);
+        if (posY > 90)
+          return;
+
+        // this part of the trigger only gets reached once per set of spheres,
+        // so we can increment the counter
+        data.iceSphereAttackCount++;
```

Comment:
(Opinion only) Is there a specific reason to place this here as opposed to in a `run` parameter? I'm not entirely against writing to `data` from within a `Text` parameter, but if it's just for saving vertical height, I feel like it's better to compartmentalize any changes to `data` within `run/preRun` if we can.

Reply:
Don't disagree stylistically, but there is a reason in this case.  The trigger fires 5 times (one for each sphere added), but we return no text for 4 of them, only caring about the one that's north of `y:90`.  If we moved this to a `run` block, it would get fired five times (unless we mirrored the same logic down there, which feels duplicative).  The other options (like using a Collect-Call trigger pair) are overkill for what we need, so this seemed like the simplest way.  Let me know if you feel differently though.

Reply:
No, that makes sense. It's hard to get a sense of what exactly is happening for this trigger without knowing how the mechanic works. Let's leave it as it is then.